commit 7c26eeff5eef526161de5365b14d323e20a4ec3f
Author: Gary Lin <glin@suse.com>
Date:   Fri Mar 30 12:25:34 2018 +0800

    efisiglist: Copy the header correctly
    
    signature_list wasn't copied corretly to efi_signature_list because
    SignatureType in signature_list is a pointer while that in
    efi_signature_list is not.
    
    Signed-off-by: Gary Lin <glin@suse.com>

diff --git a/src/siglist.c b/src/siglist.c
index 6e59164..e74eb10 100644
--- a/src/siglist.c
+++ b/src/siglist.c
@@ -217,7 +217,10 @@ signature_list_realize(signature_list *sl, void **out, size_t *outsize)
 		return -1;
 	esl = ret;
 
-	memcpy(esl, sl, sizeof (*esl));
+	memcpy(&esl->SignatureType, sl->SignatureType, sizeof(efi_guid_t));
+	esl->SignatureListSize = sl->SignatureListSize;
+	esl->SignatureHeaderSize = sl->SignatureHeaderSize;
+	esl->SignatureSize = sl->SignatureSize;
 
 	uint8_t *pos = ret + sizeof (*esl);
 	for (int i = 0; i < count; i++) {
